src/ispc.cpp

```diff
@@ -260,160 +204,37 @@ static ISPCTarget lGetSystemISA() {
-            [[maybe_unused]] bool avx10_1_512 =         avx10_512 && (avx10_ver >= 1);
-            [[maybe_unused]] bool avx10_2_256 =         avx10_256 && (avx10_ver >= 2);
-            [[maybe_unused]] bool avx10_2_512 =         avx10_512 && (avx10_ver >= 2);
-            // clang-format on
-
-            // Diamond Rapids:         DMR = GNR + AVX10_2_512 + APX + ... (For the whole list see x86TargetParser.cpp)
-            bool dmr = gnr && avx10_2_512 && apx && cmpccxadd && avxneconvert && avxifma && avxvnniint8 &&
-                       avxvnniint16 && amxcomplex && sha512 && sm3 && sm4;
-            if (dmr) {
-                return ISPCTarget::avx10_2_512_x16;
-            }
-        }
-
-        // clang-format on
-        if (spr) {
-            // We don't care if AMX is enabled or not here, as AMX support is not implemented yet.
-            return ISPCTarget::avx512spr_x16;
-        } else if (icl) {
-            return ISPCTarget::avx512icl_x16;
-        } else if (skx) {
-            return ISPCTarget::avx512skx_x16;
-        } else if (knl) {
-            Error(SourcePos(), "Detected unsupported KNL ISA.  Exiting.");
-            exit(1);
-        }
-        // If it's unknown AVX512 target, fall through and use AVX2
-        // or whatever is available in the machine.
-    }
-
-    if (osxsave && avx && __os_has_avx_support()) {
-        if (avx_vnni) {
-            return ISPCTarget::avx2vnni_i32x8;
-        }
-        // AVX1 for sure....
-        // Ivy Bridge?
-        if (avx_f16c && avx_rdrand && avx2) {
-            return ISPCTarget::avx2_i32x8;
-        }
-        // Regular AVX
-        return ISPCTarget::avx1_i32x8;
-    } else if (sse42) {
-        return ISPCTarget::sse4_i32x4;
-    } else if (sse41) {
-        return ISPCTarget::sse41_i32x4;
-    } else if (sse2) {
+    enum Target::ISA isa = (enum Target::ISA)dispatch::get_x86_isa();
+    switch (isa) {
+    case Target::ISA::SSE2:
         return ISPCTarget::sse2_i32x4;
-    } else {
-        Error(SourcePos(), "Unable to detect supported SSE/AVX ISA.  Exiting.");
+    case Target::ISA::SSE41:
+        return ISPCTarget::sse41_i32x4;
+    case Target::ISA::SSE42:
+        return ISPCTarget::sse4_i32x4;
+    case Target::ISA::AVX:
+    case Target::ISA::AVX11:
+        return ISPCTarget::avx1_i32x8;
+    case Target::ISA::AVX2:
+        return ISPCTarget::avx2_i32x8;
+    case Target::ISA::AVX2VNNI:
+        return ISPCTarget::avx2vnni_i32x8;
+    case Target::ISA::KNL_AVX512:
+        Error(SourcePos(), "Detected unsupported KNL ISA. Exiting.");
+        exit(1);
+    case Target::ISA::SKX_AVX512:
+        return ISPCTarget::avx512skx_x16;
+    case Target::ISA::ICL_AVX512:
+        return ISPCTarget::avx512icl_x16;
+    case Target::ISA::SPR_AVX512:
+        return ISPCTarget::avx512spr_x16;
+    case Target::ISA::AVX10_2_512:
+        return ISPCTarget::avx10_2_512_x16;
+    default:
+        Error(SourcePos(), "Detected unsupported KNL ISA. Exiting.");
```

Comment:
Default error message should be updated.

Reply:
Done